library/src/network.rs

```diff
@@ -8,24 +8,44 @@ use std::path::Path;
 use std::string::ToString;
 
 use crate::cache::UpdaterState;
-use crate::config::{current_arch, current_platform, ResolvedConfig};
+use crate::config::{current_arch, current_platform, UpdateConfig};
 
 // https://stackoverflow.com/questions/67087597/is-it-possible-to-use-rusts-log-info-for-tests
 #[cfg(test)]
 use std::println as info; // Workaround to use println! for logs.
 
-#[cfg(test)]
-use crate::config::{with_thread_config, with_thread_config_mut};
-
 fn patches_check_url(base_url: &str) -> String {
     return format!("{}/api/v1/patches/check", base_url);
 }
 
-#[cfg(test)]
 pub type PatchCheckRequestFn = fn(&str, PatchCheckRequest) -> anyhow::Result<PatchCheckResponse>;
+pub type DownloadFileFn = fn(&str) -> anyhow::Result<Vec<u8>>;
+
+#[derive(Clone)]
+pub struct NetworkHooks {
```

Comment:
Documentation would be helpful here. "A container for network request callbacks, used for overriding default behavior in tests"